### DISCUSSION TITLE
Feat: Side panel to adjust model parameters and refresh models (currently request a lot of clicks)

### DISCUSSION THREAD

**Discussion by sammcj** [open-webui/open-webui#3619] [Ideas]
I really love Open-WebUI, but as a frequent user I find the number of clicks I have to do to change basic parameters and refresh models to be a bit cumbersome. I would love to see a feature where I can change the parameters of a model and refresh the models from the main chat window.

Please do not think this is a complaint, I am very grateful for the work that has gone into Open-WebUI and I am very happy with the current functionality. I just think this would be a nice quality of life improvement.

## The problem

For context, I am using the dev branch, I'm aware the Admin Panel is being worked on there, but I figured my request would be more applicable to the near future.

### Refreshing models

Currently to refresh models, it takes around **11-12 clicks**.

1. The user opens the chat interface.
2. The user searches for the model and realises it's not there.
3. The user has to click their profile icon / name.
4. The user has to click the 'Admin Panel' button.
5. The user has to click the 'Settings' tab.
6. The user has to click the 'Connections' tab.
7. The user has to find the server they want to refresh the models from and click the 'Verify Connection' button.
8. The user has to click to get back to the new chat interface.
9. The user has to click search.
10. The user has to search for the model again.
11. The user has to click the model name.
12. The user has to click the 'Send a Message' text input field.

This feels like a lot of clicks 😂, it's also somewhat jarring navigating the various UI layouts (search, profile, admin horizontal tabs, admin vertical tabs).

### Changing parameters

A large part of working with LLMs is frequently adjusting the parameters of the model and model server.

Currently, to adjust basic parameters such as num_ctx, temperature, top_k, etc. it takes around **12-13 clicks**.

1. The user opens the chat interface.
2. The user searches for the model.
3. The user clicks the model name.
4. The user realises there's no way to change the parameters (or maybe this is just me 😅).
5. The user has to click their profile icon / name.
6. The user has to click the 'Settings' button.
7. The user has to search for the model settings/parameters, realises they're buried in 'Advanced Parameters' and has to click 'Show'.
8. The user has to scroll down to see the parameters.
9. The user has to click on the parameter they want to change and update the value.
10. The user has to click 'Save'.
11. The user has to click 'X' to close the settings modal.
12. The user has to click the 'Send a Message' text input field.
13. The user has to repeat for each time they want to tweak a parameter.

This is also a lot of clicks, I'll admit it actually drives me a bit crazy 😂.

## Suggestion

I think both these problems could be resolved and the user experience improved in general by having a collapsible right hand pane that shows the parameters of the model and allows the user to change them or refresh available models.

Additionally the 'Send a Message' input field could perhaps take the default focus in most situations?

I have included a shitty mock for my idea:

### Shitty Mock

![side-panel-shitty-mock](https://github.com/open-webui/open-webui/assets/862951/4128c305-a284-4a3f-a992-7a91be5e6a2c)

---

I would submit a PR to add this functionality but my web UI skills for something like this are definitely not up to par, however I would of course be more than happy to help test this feature if someone else is willing to have a go at implementing it.

Thanks for considering this feature request!

---

### Tweaking parameters

![parameters-1](https://github.com/open-webui/open-webui/assets/862951/ed6181e8-417a-483e-9129-bf93fa690c77)

![parameters-2](https://github.com/open-webui/open-webui/assets/862951/85ca116d-466e-470e-ab09-e96284374345)

![parameters-3](https://github.com/open-webui/open-webui/assets/862951/4f89451f-037e-433f-aba2-32aa577ba1c8)

![parameters-4](https://github.com/open-webui/open-webui/assets/862951/768b3819-cfa1-4baa-afb6-53ef0988410f)

![parameters-5](https://github.com/open-webui/open-webui/assets/862951/0f30d1cc-4429-4be8-b65a-a5337825d35f)

![parameters-6](https://github.com/open-webui/open-webui/assets/862951/1e814c48-0997-4d2a-bfa5-f785628f1166)

![parameters-7](https://github.com/open-webui/open-webui/assets/862951/d9e76f3b-bb37-4341-9246-9ee9df73acb4)

![parameters-8](https://github.com/open-webui/open-webui/assets/862951/71bb7f1e-7324-4cf9-9511-b8e257c81b65)

![parameters-9](https://github.com/open-webui/open-webui/assets/862951/04b4e889-4f52-4f63-bf73-53ad724eff97)

### Refreshing models

![refresh-models-1](https://github.com/open-webui/open-webui/assets/862951/b79d54c6-319f-4572-b8a8-ff3112c9acf7)

![refresh-models-2](https://github.com/open-webui/open-webui/assets/862951/db2be9fb-eb50-4676-9e14-221e41169440)

![refresh-models-3](https://github.com/open-webui/open-webui/assets/862951/51443e62-59c1-48f0-90a7-bc07ca52fdca)

![refresh-models-4](https://github.com/open-webui/open-webui/assets/862951/f06b3d3f-233b-4b2d-bdcc-62df606f0280)

![refresh-models-5](https://github.com/open-webui/open-webui/assets/862951/90043acf-ae06-4dde-a73f-276bbf1f69ce)

![refresh-models-6](https://github.com/open-webui/open-webui/assets/862951/cd92f56a-482e-43e8-a936-34a42878e5fa)

![refresh-models-7](https://github.com/open-webui/open-webui/assets/862951/a1b8eff5-a889-4d92-bc4f-ee8cc9c993fa)

![refresh-models-8](https://github.com/open-webui/open-webui/assets/862951/4b56bb38-a674-4af6-acff-9f199463cfeb)

**Comment by tjbck** [comment]
Never say never but the right-side panel will not be added. However, I'll see if we can try to reduce the number of clicks required for adjusting the params. Related: #619

**Reply by sammcj** [reply]
Aw, that's a shame. What's the reason behind this out of interest?

They're really useful with interfaces like LM-Studio, InvokeAI etc...

**Reply by tjbck** [reply]
Having two side panels looks too clustered for my liking, but I do agree it does make tweaking the parameters a lot easier. Again, I might just end up with the same conclusion and decide having two panels is the best solution but I'd like to keep our core design as minimalistic and clean as possible. This has been on my mind for a while now so I'll try to take a look in a bit. Stay tuned!

**Reply by sammcj** [reply]
The other way I could imagine doing it would be to have a fold out (up) panel from the text input field that contains the parameters, it could auto-hide when a message is sent.

**Comment by tjbck** [comment]
As for the refreshing the list of models, you can just refresh the window (F5).

**Comment by tjbck** [comment]
![image](https://github.com/open-webui/open-webui/assets/25473318/f274ae95-e6bc-422f-af7e-d5e020418ecc)


Per chat config feature has been added to our dev branch as "Chat Controls"!

**Reply by sammcj** [reply]
Oh my! I love it! Thank you so much!

**Reply by sammcj** [reply]
Using 0.38.0 with this - it's a huge improvement, thank you, excellent work!

**Reply by sammcj** [reply]
Using this every single day - it's honestly meant I pick open-webui over some other tools _far_ more often! Thanks again @tjbck !
